[gui] Show filters in expansion panels

csordasmarton opened the pull request:
We already have many report filters on the UI and it is hard to see
through where is the filter what the user actually want to set. For
this reason with this patch we will show each filter in an expansion
panel and every filter will be closed by default. We will also show
the selected filter items in the filter header bar which will be trimmed
when too many filter items are selected.

Screenshot:
![image](https://user-images.githubusercontent.com/6695818/89635721-b7064180-d8a7-11ea-8f84-cf21cb1fb125.png)

bruntib commented:
Wow, this looks great!
Could Baseline and Newcheck be visually distinguished somehow? In the screenshot it looks like "Baseline" and "Newcheck" are field groups and "Run filter", "Tag filter" belong to "Baseline", every other filter options belong to "Newcheck". This is purely a visual concern.

csordasmarton commented:
@bruntib I moved the expansion icon to the left for BASELINE and NEWCHECK filter section. I think it will look much better now. I updated the screenshot above. What do you think?

gyorb commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=5968187&bid=19567703)

gyorb commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=5968187&bid=19567703)

dkrupp commented:
Can you differentiate the diffing filters even a bit more? Like color baseline and newcheck to different color compared to the other filters?

Also could you please rename "NEWCHECK" to "Compare to". I think it would make a lot more sense to people.

csordasmarton commented:
@dkrupp I added a border to the diff sections (baseline/newcheck). I updated the screenshot above. Is it better now?

I do the renaming in another pull request: #2871.

gyorb commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=5968187&bid=19567703)

dkrupp commented:
> @dkrupp I added a border to the diff sections (baseline/newcheck). I updated the screenshot above. Is it better now?
> 
> I do the renaming in another pull request: #2871.

Yes looks good now, thanks.

gyorb commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=5968187&bid=19567703)
